### PR TITLE
overlord/state, daemon: add Change ready channel

### PR DESCRIPTION
**Pull request by niemeyer**
Depends on task-undo (#745).

May be reviewed with https://github.com/ubuntu-core/snappy/pull/746/commits/fc6dddaf085a080f4113fbca5cf996c79a05a77b.

### CODE REVIEW

File: daemon/api.go [hunk 1]
```diff
@@ -641,17 +641,13 @@ func (inst *snapInstruction) Agreed(intro, license string) bool {
...
-		st.Lock()
-		s := chg.Status()
-		if s == state.DoneStatus || s == state.ErrorStatus {
-			defer st.Unlock()
-			return chg.Err()
-		}
-		st.Unlock()
-		time.Sleep(250 * time.Millisecond)
+	select {
+	case <-chg.Ready():
```
Comment (mvo5) [inline daemon/api.go]: Nice!

File: overlord/state/change.go [hunk 2]
```diff
@@ -171,6 +173,7 @@ func (c *Change) UnmarshalJSON(data []byte) error {
 	c.status = unmarshalled.Status
 	c.data = unmarshalled.Data
 	c.taskIDs = unmarshalled.TaskIDs
+	c.ready = make(chan struct{})
```
Comment (pedronis) [inline overlord/state/change.go]: I think, to be consistent, we should close c.ready here if status is ready, something to test in the checkpointing tests

Comment (niemeyer) [inline overlord/state/change.go]: Of course, sorry about that.

**Comment by pedronis** [comment]
LGTM

**Pull request merged**